### DISCUSSION TITLE
How to add widgets to ScrolledFrame?

### DISCUSSION THREAD

**Discussion by ponchofiesta** [alejandroautalan/pygubu-designer#170] [Q&A] [answered]
I want to add child elements to a ScrolledFrame. This works statically in the pygubu-designer. But how to add widgets programmatically?

```python
# This is the ScrolledFrame widget
groups_frame = builder.get_object("container")
# Trying to add custom widget to ScrolledFrame
group = ImageGroup(...., root=self.groups_frame)
```
The ImageGroup widget is shown but below the ScrolledFrame. Do I need another root as parent widget?

**Comment by alejandroautalan** [comment, marked as answer]
Hello @ponchofiesta thanks for trying pygubu.

> I want to add child elements to a ScrolledFrame. This works statically in the pygubu-designer. But how to add widgets programmatically?

Create the child widgets using the scrolledframe.**innerframe**  property as master.
In your example:

```python
# This is the ScrolledFrame widget
groups_frame = builder.get_object("container")
# Trying to add custom widget to ScrolledFrame
group = ImageGroup(...., root=self.groups_frame.innerframe)
```
Let me know
Regards

**Reply by ponchofiesta** [reply]
Works perfectly. Thank you.
